data/onPostBuild/generateMarkdown.ts

```diff
@@ -0,0 +1,210 @@
+import { GatsbyNode } from 'gatsby';
+import * as path from 'path';
+import * as fs from 'fs';
+import TurndownService from 'turndown';
+import cheerio from 'cheerio';
+import fastGlob from 'fast-glob';
+
+/**
+ * This script generates Markdown static files alongside HTML files for each page.
+ * This allows LLM crawlers and other clients to request markdown versions of pages
+ * which are significantly more token-efficient than HTML.
+ */
+
+const REPORTER_PREFIX = 'generateMarkdown:';
+
+// Constants for content validation
+const REDIRECT_PAGE_MAX_SIZE = 1000; // Maximum size in bytes for redirect pages
+const MIN_CONTENT_LENGTH = 100; // Minimum content length to consider meaningful
+
+// Configure Turndown for documentation-friendly markdown
+const createTurndownService = () => {
+  const turndownService = new TurndownService({
+    headingStyle: 'atx',
+    codeBlockStyle: 'fenced',
+    bulletListMarker: '-',
+    emDelimiter: '_',
+    strongDelimiter: '**',
+    linkStyle: 'inlined',
+    linkReferenceStyle: 'full',
+  });
+
+  // Preserve code block language annotations
+  turndownService.addRule('fencedCodeBlock', {
```

Comment:
Not sure if the related bit, but we should scrub out code block lines in generated MD. We get nice blocks surrounded by triple back-ticks, but then a series of numbers 1-N before it, each on their own lines

Reply:
Yeah this is the one we just need to nail down. My next move in #2657 was to solve this, and I was going to go down the path of generating files with double extensions, ie `thing.javascript.md` or `thing.swift.md`, rather than using `-`.

The other thing is that it is not just code blocks, Mark can use a component to restrict pieces of copy to only show up under certain languages... My (untested) idea was to load up the page with jsdom and manipulate the selected language so the _whole_ DOM is updated correctly before passing the markup to Turndown...

Reply:
@mattheworiordan I'm not sure if you saw my earlier comment here